include/seqan3/search/fm_index/bi_fm_index.hpp

```diff
@@ -28,75 +28,30 @@ namespace seqan3
  * \{
  */
 
-/*!\brief The default Bidirectional FM Index Configuration.
- *
- * \details
- *
- * \todo write me
- *
- */
-struct bi_fm_index_default_traits
-{
-    //!\brief Type of the underlying forward SDSL index.
-    using fm_index_traits = fm_index_default_traits;
-
-    //!\brief Type of the underlying reverse SDSL index.
-    using rev_fm_index_traits = fm_index_default_traits; // TODO: trait object without sampling.
-};
-
 /*!\brief The SeqAn Bidirectional FM Index
  * \implements seqan3::BiFmIndex
- * \tparam text_t The type of the text to be indexed; must model std::ranges::RandomAccessRange.
- * \tparam bi_fm_index_traits The traits determining the implementation of the underlying SDSL indices;
-                              must model seqan3::BiFmIndexTraits.
+ * \tparam is_collection_ Indicates whether this index works on a text collection (`true`) or a single text (`false`)
+ * \tparam sdsl_index_type_ The type of the underlying SDSL index, must model seqan3::SdslIndex.
  * \details
  *
  * \todo write me
  */
-template <std::ranges::RandomAccessRange text_t, BiFmIndexTraits index_traits_t = bi_fm_index_default_traits>
-//!\cond
-    requires Semialphabet<innermost_value_type_t<text_t>> &&
-             std::Same<alphabet_rank_t<innermost_value_type_t<text_t>>, uint8_t>
-//!\endcond
+template <bool is_collection_ = false, detail::SdslIndex sdsl_index_type_ = sdsl_index_default_type>
 class bi_fm_index
 {
-protected:
-    //!\privatesection
-
-    //!\brief Pointer to the indexed text.
-    text_t const * text = nullptr;
-
 public:
-
-    static_assert(dimension_v<text_t> == 1 || dimension_v<text_t> == 2,
-                  "Only texts or collections of texts can be indexed.");
-
     //!\brief Indicates whether index is built over a collection.
-    static bool constexpr is_collection = dimension_v<text_t> == 2;
-
-    /*!\name Text types
-     * \{
-     */
-    //!\brief The type of the forward indexed text.
-    using text_type = text_t;
-    //!\brief The type of the forward indexed text.
-    using rev_text_type = std::conditional_t<is_collection,
-                                             decltype(*text | view::deep{std::view::reverse} | view::deep{view::persist}
-                                                            | std::view::reverse),
-                                             decltype(*text | std::view::reverse)>;
-    //!\}
+    static constexpr bool is_collection{is_collection_};
```

Comment:
if it is private, I would rather have the `_` here and the well-readable one in the signature...

Reply:
I'll make this one private and switch the names